### PR TITLE
Decoupling & Progressbar (Oops)

### PR DESCRIPTION
**Pull request by kingce**
Initial feature branch where we moved a class into it's own file and included code for the progress bar (feature OK'd by Clauson in meeting on 4/30 for reference). 

I know these definitely should've been in two separate branches but since this work was peer-programmed and mostly done synchronously and we didn't want to deal with merge conflicts with moving the ExamImage class after inserting code into it.

Once this is merged we'll stick to the 'one feature per branch' workflow - maybe need to discuss how branching works in the context of the ongoing decoupling/structure changes.

### CODE REVIEW

File: program/Makefile [hunk 1]
```diff
@@ -1,2 +1,995 @@
-PDFGrader.exe: src/main/System.vala src/main/QuestionSet.vala src/main/Question.vala src/main/Mark.vala src/main/Save.vala
-	valac -o PDFGrader.exe --pkg=gtk+-3.0 --pkg=poppler-glib --pkg cairo --pkg gee-0.8 --pkg glib-2.0 --pkg libxml-2.0 --pkg posix src/main/System.vala src/main/QuestionSet.vala src/main/Question.vala src/main/Mark.vala src/main/Save.vala src/main/Lock.vala src/main/Export.vala
+# Makefile.in generated by automake 1.15.1 from Makefile.am.
```
Comment (MaxLisaius) [inline program/Makefile]: Why are we replacing the Makefile with an auto-generated one?

Comment (kingce) [reply program/Makefile]: Really not sure how this snuck its way in, don't think anyone here made changes to this file on purpose. Is there a way to revert just this one file to the source branch?

Comment (kingce) [reply program/Makefile]: Ran 'git checkout develop -- program/Makefile' to revert the makefile to what it is on develop. Lmk if this isn't correct/should be done in a different way.

File: program/Makefile.am [hunk 2]
```diff
@@ -9,6 +9,7 @@ pdfgrader_LDADD = $(pdfgrader_LIBS)
 pdfgrader_VALAFLAGS = --pkg=gtk+-3.0 --pkg=poppler-glib --pkg cairo --pkg gee-0.8 --pkg glib-2.0 --pkg libxml-2.0 --pkg posix
 
 pdfgrader_SOURCES = src/main/System.vala \
+    src/main/ExamImage.vala \
```
Comment (clemwj) [inline program/Makefile.am]: Different indent

File: program/src/main/System.vala [hunk 3]
```diff
@@ -40,17 +41,14 @@ public class System
     }
 
     //sets up gtk stuff
-    private static void initUI()
-    {
+    private static void initUI() {
```
Comment (clemwj) [inline program/src/main/System.vala:44]: Love the starting of the updating. Thanks!

File: program/src/main/System.vala [hunk 4]
```diff
@@ -101,35 +99,46 @@ public class System
...
+
         //setup window area UI
         {
             //setup the exam view on left, and tools on right
             var hpaned = new Gtk.Paned(Gtk.Orientation.HORIZONTAL);
             hpaned.set_size_request(800, 400); //minimum size of entire window content
             menuPane.pack2(hpaned, true, false);
             
-            //left side: exam view
+            //left side: exam view AND progressBar?
```
Comment (clemwj) [inline program/src/main/System.vala]: Should this question mark be here?

File: program/Makefile.am (hunk 2 above)
Comment (kingce) [reply program/Makefile.am]: For some reason this didn't show up on my local code editor. Fixed it directly from github - thanks for pointing this out!

File: program/src/main/System.vala [hunk 5]
```diff
@@ -101,35 +99,46 @@ public class System
...
+            var viewPane = new Gtk.Paned(Gtk.Orientation.VERTICAL);
```
Comment (clemwj) [inline program/src/main/System.vala]: This file hasn't been updated for coding conventions so i don't expect you to do the work to update it. But new variable names should be lowercase_with_underscores

File: program/src/main/System.vala [hunk 6]
```diff
@@ -204,6 +213,9 @@ public class System
         var questionSet = examQuestionSet.get(0); //since the first entry is the name space, we need to add 1
         var currQuestion = questionSet.getQuestions().get(currentTest); //the specific student's test question
         var pool = questionSet.rubricPool;
+
+        progressBar.set_fraction(5/10); //TEST
```
Comment (clemwj) [inline program/src/main/System.vala]: Test code

File: program/src/main/System.vala [hunk 7]
```diff
@@ -294,6 +306,8 @@ public class System
         var currQuestion = questionSet.getQuestions().get(currentTest); //the specific student's test question
         var questionActiveMarks = currQuestion.getMarks();
         
+        progressBar.set_fraction(currentTest/2); //PROGRESS_BAR_TEST
```
Comment (clemwj) [inline program/src/main/System.vala]: Same thing here

File: program/src/main/System.vala (hunk 4 above)
Comment (kingce) [reply program/src/main/System.vala]: Oops, looks like a few of my testing comments went through. Will update all comments in this repo this afternoon - sorry about that! Definitely should've gone through this again before opening the PR

File: program/src/main/System.vala [hunk 8]
```diff
@@ -738,7 +752,9 @@ public class System
         }
 
         // load the new question from the file.
-        QuestionSet nextQuestion;
+        QuestionSet nextQuestion; //note 1- update progress bar here
+        progressBar.set_fraction(9/10); //PROGRESS_BAR_TEST
```
Comment (clemwj) [inline program/src/main/System.vala]: Do you need to do something based on this comment? And another test

File: program/src/main/System.vala [hunk 9]
```diff
@@ -862,956 +878,3 @@ public class MarkViewWorthEntry : Gtk.Entry
     }
 }
 
-//Class that handles all of the rendering of a pdf document page.
```
Comment (clemwj) [inline program/src/main/System.vala:865]: Good job with this. I'm glad that it wasn't too difficult

File: program/src/main/System.vala (hunk 8 above)
Comment (kingce) [reply program/src/main/System.vala]: I think that comment was made before we changed the progressbar design to be simply based on keypress updates. Removed!

**Review by clemwj — APPROVED** [review]

**Pull request merged**